### ISSUE TITLE
Reopen #1919 & #2291: Bottom part of ionic tabs is not clickable

### ISSUE THREAD

**Issue by KiwiKilian** [mapsplugin/cordova-plugin-googlemaps#2409]
**I'm submitting a ...**  (check one with "x")
- [ ] question
- [x] any problem or bug report

**OS:**  (check one with "x")
- [ ] **Android**
- [x] **iOS**
- [ ] **Browser**

**cordova information:**  (run `$> cordova plugin list`) 

```
com.googlemaps.ios 2.7.0 "Google Maps SDK for iOS"
cordova-plugin-device 2.0.2 "Device"
cordova-plugin-googlemaps 2.4.1 "cordova-plugin-googlemaps"
cordova-plugin-ionic-keyboard 2.1.2 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 2.1.0 "cordova-plugin-ionic-webview"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-statusbar 2.4.2 "StatusBar"
cordova-plugin-whitelist 1.3.3 "Whitelist"
```

**If you use `@ionic-native/google-maps`, please tell the package.json (only `@ionic-native/core` and `@ionic-native/google-maps` are fine mostly)**
```
@ionic-native/core : 4.12.0
@ionic-native/google-maps : 4.12.0
```

**Current behavior:**
Bottom part of Ionic tabs are not clickable when Cordova Google Maps is installed.
Same issue as described in #1919 and #2291, seems still not fixed. When Cordova Google Maps is installed the tabs are only clickable in the upper two thirds, as it can be seen in the following video.

<img src="https://user-images.githubusercontent.com/18724559/44875194-be754700-ac9d-11e8-8fdc-56817e8c8eb8.gif" width="250">

**Expected behavior:**
Bottom part of Ionic tabs clickable. For testing I completely removed the Cordova Google Maps Plugin. Afterwards the tabs are clickable everywhere, just as expected. See the second video below.

<img src="https://user-images.githubusercontent.com/18724559/44875195-be754700-ac9d-11e8-900b-474ea35995ed.gif" width="250">

**Repository:**
[Repository with non working example](https://github.com/KiwiKilian/ionic-google-maps-debug)

**Comment by wf9a5m75** [comment]
I just built and run your repo.
![2409](https://user-images.githubusercontent.com/167831/44878436-bac0d080-ac5b-11e8-92b6-4221f8cbd78f.gif)

**Comment by KiwiKilian** [comment]
Emphasize lays on **bottom** part. In case of the tabs that's the area where the text is, which is not clickable. You are clicking on the icons which works fine indeed. It's also easier to detect on any device except iPhone-X. Therefore I please use `ionic cordova emulate ios --target=iPhone-8`.

I updated my [demo](https://github.com/KiwiKilian/ionic-google-maps-debug) to make very clear, which area is not clickable, also the problem is not limited to tabs. Any element in the very lowest ~18px-20px  at the bottom of the screen (red area) is affected. The alert shows "Upper" when the green area is clicked and "Lower" when the red area is clicked. As you can see, the red area is not clickable on iOS.

![improved demo](https://user-images.githubusercontent.com/18724559/44897393-cc59b500-acfb-11e8-8c2b-223e1d8c7f96.gif)

As proof, here is Android doing just fine:

![android](https://user-images.githubusercontent.com/18724559/44897677-d03a0700-acfc-11e8-840b-c3d1a99b7cca.gif)

It just seems like a very small area that's not clickable but it's really annoying on a physical device when selecting tabs.
This might be again related to miscalculation with the statusbar?
Really appreciate your effort and help!

**Comment by wf9a5m75** [comment]
Do you create a small button such in the super bottom area?

**Comment by wf9a5m75** [comment]
Then removing `cordova-plugin-statusbar` is fine. I think they shiftdownes the webView mandatory.

**Comment by KiwiKilian** [comment]
Well `cordova-plugin-statusbar` is installed in nearly every ionic/cordova app. I think removing this plugin as many people will need to modify the look of the statusbar is not a real solution and this should be considered a bug.

**Comment by wf9a5m75** [comment]
Well, I didn't remove it. I just created the project with tutorial steps, and I didn't know the status bar plugin is not included.
Are you sure ionic install 100% the status bar plugin for all projects?


Anyway, if you can't satisfy the bottom of 20px, please create a pull request.
Click detection is `hitTest` in this file.
https://github.com/mapsplugin/cordova-plugin-googlemaps/blob/master/src/ios/GoogleMaps/MyPluginLayer.m#L322

I will test your repo tomorrow if I have time.

**Comment by KiwiKilian** [comment]
I just checked with Ionic Starters Tabs and it's currently only in a half working state. They are calling methods of the `cordova-plugin-statusbar` but don't completely install it. I filed an issue with them to confirm if they plan to keep this plugin as default (https://github.com/ionic-team/starters/issues/430).

I will try looking into the `hitTest` myself. Would be awesome if you also have a shot at my new demo and confirm that we actually have a bug/problem when using both plugins together. Thank you very much again!

**Comment by wf9a5m75** [comment]
I confirmed this is **NOT** this plugin's fail.

This is the view hierarchy of application.
There is an UIImageView at the very bottom of application (blue frame).
That's the `cordova-plugin-statusbar`.

![screen shot 2018-08-31 at 11 51 35 am](https://user-images.githubusercontent.com/167831/44930788-49912400-ad14-11e8-9c38-ef7a45a25a74.png)

This plugin catches all touch event at the very first.
Then, in order to work with other native plugins (such as `PhoneGap-Plugin-ListPicker`),  asks other views the touch is necessary for them or not.
If one of them says `yes, I need this touch (return something from [subview hitTest:point withEvent:event];)`, this plugin passes the event.

https://github.com/mapsplugin/cordova-plugin-googlemaps/blob/730f5a7cc37a61687e84bd3e17f402a84bd2f469/src/ios/GoogleMaps/MyPluginLayer.m#L329-L353

Here is the log.

```
2018-08-31 12:09:00.945062-0700 ionic-google-maps-debug[9193:110687] --->subview[2] = <UIScrollView: 0x7fe513813600; frame = (0 0; 375 667); clipsToBounds = YES; gestureRecognizers = <NSArray: 0x60400025e510>; layer = <CALayer: 0x6000000337c0>; contentOffset: {0, 647}; contentSize: {375, 1334}; adjustedContentInset: {20, 0, 0, 0}>
2018-08-31 12:09:00.945516-0700 ionic-google-maps-debug[9193:110687] --->hit = <UIScrollView: 0x7fe513813600; frame = (0 0; 375 667); clipsToBounds = YES; gestureRecognizers = <NSArray: 0x60400025e510>; layer = <CALayer: 0x6000000337c0>; contentOffset: {0, 647}; contentSize: {375, 1334}; adjustedContentInset: {20, 0, 0, 0}>
2018-08-31 12:09:00.946231-0700 ionic-google-maps-debug[9193:110687] --->subview[2] = <UIScrollView: 0x7fe513813600; frame = (0 0; 375 667); clipsToBounds = YES; gestureRecognizers = <NSArray: 0x60400025e510>; layer = <CALayer: 0x6000000337c0>; contentOffset: {0, 647}; contentSize: {375, 1334}; adjustedContentInset: {20, 0, 0, 0}>
2018-08-31 12:09:00.946715-0700 ionic-google-maps-debug[9193:110687] --->hit = <UIScrollView: 0x7fe513813600; frame = (0 0; 375 667); clipsToBounds = YES; gestureRecognizers = <NSArray: 0x60400025e510>; layer = <CALayer: 0x6000000337c0>; contentOffset: {0, 647}; contentSize: {375, 1334}; adjustedContentInset: {20, 0, 0, 0}>
```

After passing the event, this plugin does not business it anymore.

-----

So if you want to put `clickable area in the very bottom area`, you need to remove the `cordova-plugin-statusbar` plugin.

**Comment by KiwiKilian** [comment]
Not using `cordova-plugin-statusbar` and `cordova-plugin-googlemaps` together is not quite a solution for most people.

The `UIImageView` is not from `cordova-plugin-statusbar`. I removed the plugin and the same `UIImageView` still appears, therefore it's not from the statusbar plugin. The file [`CDVStatusBar.m`](https://github.com/apache/cordova-plugin-statusbar/blob/master/src/ios/CDVStatusBar.m) also doesn't reference any `UIImageView`. It also doesn't cover the non-clickable area completely, therefore I highly doubt that this `UIImageView` is related to the problem at all.

I looked into the code, but couldn't yet find a fix. But you did fix the problem before in https://github.com/mapsplugin/cordova-plugin-googlemaps/commit/d41f4362df341161a5119a46b6d979e9c2aa2e3b. I downgraded my test app to `cordova-plugin-googlemaps` v2.2.1 and the problem is fixed indeed (see [this branch](https://github.com/KiwiKilian/ionic-google-maps-debug/tree/test/old-version)). Therefore the changes introduced afterwards must have reintroduced the problem and so I don't feel comfortable calling it "not a bug".

**Comment by wf9a5m75** [comment]
I ask again, `Do you create a small button such in the super bottom area?`

**Comment by KiwiKilian** [comment]
> Then you should send a PR, that's open source community contribution.
I'm looking into it, but don't understand enough of it yet to be able to fix it.

> Do you create a small button such in the super bottom area?
I'm not quite sure what you want to ask with this wording.
This setup is just for demonstration, in our real application the bug limits the clickable area of the Ionic Tabs resulting in bad user experience. I created the green and red areas with <div>s which are clickable. The html code is [here](https://github.com/KiwiKilian/ionic-google-maps-debug/blob/master/src/pages/home/home.html).

**Comment by wf9a5m75** [comment]
wow, your finger might be really sharp.

I don't feel any stress when I tab application on iPhone because my finger is bigger than 20px.

------

I'm not sure who insert the `UIImageView`s.
You said the `cordova-plugin-statusbar`, I said so that.
I guess the UIImageView is aim to catch the gesture that open the os controls.

As I explained, in order to work with othet native views, this plugin asks the event is necessary or not to them.
If one of them is "yes", the maps plugin does not manage it anymore.
I won't change this implementation.

If you prefer v2.2.1, please go back to v2.2.1, or folk the repo, and replace the `hitTest` with the code of v2.2.1.

**Comment by KiwiKilian** [comment]
> wow, your finger might be really sharp.
I don't feel any stress when I tab application on iPhone because my finger is bigger than 20px.

Well then phones should be only use the last 20px for display without any touch capability 😉. It actually lets you miss tabs sometimes where it normally would trigger a click.

Anyway, I think I've fixed it, see #2411.

**Comment by wf9a5m75** [comment]
Merged. Thanks.
